-------------------------------------------------------------------------------------------------------------------------
Starting new testcase :TC01_create_schedule_release_Order
Pre-condition for Test: Mock Confirm Shipment
Calling POST request with end point : http://localhost:7001/smcfs/restapi/invoke/confirmShipment
API Request sent : 
Request method:	POST
Request URI:	http://localhost:7001/smcfs/restapi/invoke/confirmShipment
Proxy:			<none>
Request params:	<none>
Query params:	<none>
Form params:	<none>
Path params:	<none>
Headers:		Accept=application/xml
				Content-Type=application/xml
Cookies:		<none>
Multiparts:		<none>
Body:
<Shipment Action="Create" DocumentType="0001" EnterpriseCode="DEFAULT" ShipNode="Matrix_WH1">
  <ShipmentLines>
    <ShipmentLine DocumentType="0001" OrderNo="" OrderReleaseKey="" PrimeLineNo="1" Quantity="2" SubLineNo="1"/>
  </ShipmentLines>
  <ShipmentLines>
    <ShipmentLine DocumentType="0001" OrderNo="" OrderReleaseKey="" PrimeLineNo="1" Quantity="2" SubLineNo="1"/>
  </ShipmentLines>
</Shipment>

API Response received : 
HTTP/1.1 400 Bad Request
Date: Mon, 08 Nov 2021 05:53:45 GMT
Content-Length: 273
Content-Type: application/xml; charset=UTF-8
Content-Encoding: gzip
X-FRAME-OPTIONS: SAMEORIGIN
X-UA-Compatible: IE=EmulateIE8

<Errors>
  <Error ErrorCode="YFS10003" ErrorDescription="YFS:Invalid Order" HttpCode="400">
    <Attribute Name="ErrorDescription" Value="YFS:Invalid Order"/>
    <Attribute Name="ErrorCode" Value="YFS10003"/>
    <Attribute Name="HttpCode" Value="400"/>
    <Attribute Name="SubLineNo" Value="1"/>
    <Attribute Name="DocumentType" Value="0001"/>
    <Attribute Name="Quantity" Value="2"/>
    <Attribute Name="OrderNo" Value=""/>
    <Attribute Name="PrimeLineNo" Value="1"/>
    <Attribute Name="EnterpriseCode" Value="DEFAULT"/>
    <Attribute Name="OrderReleaseKey" Value=""/>
  </Error>
</Errors>

Test failed : TC01_create_schedule_release_Order
java.lang.AssertionError: Status code for confirmShipment matches with expected code expected [200] but found [400]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
	at org.testng.Assert.assertEquals(Assert.java:122)
	at org.testng.Assert.assertEquals(Assert.java:907)
	at org.testng.asserts.Assertion$26.doAssert(Assertion.java:338)
	at org.testng.asserts.Assertion.executeAssert(Assertion.java:25)
	at org.testng.asserts.Assertion.doAssert(Assertion.java:12)
	at org.testng.asserts.Assertion.assertEquals(Assertion.java:334)
	at com.perfaware.automation.oms.sterling.common.applicationHelper.RequestHelper.requestCreationInOMS(RequestHelper.java:159)
	at com.perfaware.automation.oms.sterling.common.testcases.RestAPI.createOrderSTK_FC(RestAPI.java:196)
	at com.perfaware.automation.oms.sterling.common.testcases.RestAPI.TC01_create_schedule_release_Order(RestAPI.java:77)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Test failed : TC01_create_schedule_release_Order
java.lang.NullPointerException
	at com.perfaware.automation.oms.sterling.common.testListeners.TestListener.onTestFailure(TestListener.java:60)
	at org.testng.internal.TestListenerHelper.runTestListeners(TestListenerHelper.java:96)
	at org.testng.internal.TestInvoker.runTestResultListener(TestInvoker.java:219)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:656)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

